### PR TITLE
Refactor index class

### PR DESCRIPTION
**Pull request by nmellado**
Uniformize Index use.
Set Index as constexpr class.

### CODE REVIEW

**Comment by nmellado** [comment]
This PR is ready for review.

**Comment by nmellado** [comment]
Fair enough

**Comment by hoshiryu** [comment]
For the Index class methods, why do we have some defined in the .hpp and others in the .inl ?

File: src/Core/Mesh/MeshTypes.hpp [hunk 1]
```diff
@@ -13,11 +14,9 @@ namespace Ra
         typedef Vector2ui Edge;
         typedef Vector3ui Triangle;
 
-        typedef uint TriangleIdx;
-        typedef uint VertexIdx;
-        typedef uint HalfEdgeIdx;
-
-        enum { InvalidIdx = uint( -1 ) };
+        using TriangleIdx = Index;
```
Comment (hoshiryu) [inline src/Core/Mesh/MeshTypes.hpp]: why `using` for these and `typedef` for the two above?

Comment (MathiasPaulin) [reply src/Core/Mesh/MeshTypes.hpp]: The question could be more general. Do we have to use the (old-style) typedef or the more C++ clean "using" for type aliasing ?
I prefer the second ... I canchange all the typedef to using if tis is OK ...

Comment (nmellado) [reply src/Core/Mesh/MeshTypes.hpp]: Yes, please.

**Pull request merged**